app/models/miq_worker.rb

```diff
@@ -219,6 +219,10 @@ def worker_settings(options = {})
     self.class.fetch_worker_settings_from_server(miq_server, options)
   end
 
+  def heartbeat_file
+    ENV["WORKER_HEARTBEAT_FILE"] || Rails.root.join("tmp", "#{guid}.hb")
```

Comment:
Morbid curiosity (and since I got pinged on this thread)?  Why?

Is an old Heartbeat file (like a stale worker record) a problem?  Not saying we shouldn't clean things up, but just curious if there are issues you have identified where cleaning these up is necessary to resolve them.